ansible/network/files/docker-compose-rpc.yml

```diff
@@ -37,30 +40,52 @@ services:
       NRIA_DISPLAY_NAME: "${NETWORK_NAME}-rpc-node-${NODE_ID}"
     restart: unless-stopped
 
-  # caddy reverse proxy with automatic tls management using let encrypt
-  caddy:
-    image: lucaslorentz/caddy-docker-proxy:latest
+  # traefik reverse proxy with automatic tls management using let encrypt
+  traefik:
+    image: traefik:v2.10
+    container_name: traefik
     restart: unless-stopped
+    command:
+      - --api=false
+      - --api.dashboard=false
+      - --providers.docker
+      - --log.level=info
+      - --entrypoints.web.address=:80
+      - --entrypoints.web.http.redirections.entryPoint.to=websecure
+      - --entrypoints.websecure.address=:443
+      - --providers.docker=true
+      - --providers.docker.exposedbydefault=false
+      - --certificatesresolvers.le.acme.email=alerts@subspace.network
+      - --certificatesresolvers.le.acme.storage=/letsencrypt/acme.json
+      - --certificatesresolvers.le.acme.tlschallenge=true
+    networks:
+      - traefik-proxy
     ports:
-      - "80:80"
-      - "443:443"
-    environment:
-      - CADDY_INGRESS_NETWORKS=subspace_default
+      - 80:80
+      - 443:443
     volumes:
-      - /var/run/docker.sock:/var/run/docker.sock:ro
-      - caddy_data:/data
+      - /var/run/docker.sock:/var/run/docker.sock
+      - ./letsencrypt:/letsencrypt
 
   archival-node:
-    image: ghcr.io/${NODE_ORG}/node:${NODE_TAG}
+    image: ghcr.io/\${NODE_ORG}/node:\${NODE_TAG}
```

Comment:
I dont think this is correct, if you escape these envs, the image will be used as is and docker download fails